node_package/src/RSCServerRoot.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import { createFromNodeStream } from 'react-on-rails-rsc/client.node';
+import { PassThrough } from 'stream';
+import type { RenderFunction, RailsContext } from './types';
+import transformRSCStream from './transformRSCNodeStream';
+import loadJsonFile from './loadJsonFile';
+import RSCPayloadContainer from './RSCPayloadContainer';
+
+declare global {
+  function generateRSCPayload(
+    componentName: string,
+    props: Record<string, unknown>,
+    railsContext: RailsContext,
+  ): Promise<NodeJS.ReadableStream>;
+}
+
+type RSCServerRootProps = {
+  componentName: string;
+  componentProps: Record<string, unknown>;
+};
+
+if (!('use' in React && typeof React.use === 'function')) {
+  throw new Error(
+    'React.use is not defined. Please ensure you are using React 18 with experimental features enabled or React 19+ to use server components.',
+  );
+}
+
+const { use } = React;
+
+const createFromReactOnRailsNodeStream = (
+  stream: NodeJS.ReadableStream,
+  ssrManifest: Record<string, unknown>,
+) => {
+  const transformedStream = transformRSCStream(stream);
+  return createFromNodeStream(transformedStream, ssrManifest);
+};
+
+const createSSRManifest = async (
+  reactServerManifestFileName: string,
+  reactClientManifestFileName: string,
+) => {
+  const [reactServerManifest, reactClientManifest] = await Promise.all([
+    loadJsonFile(reactServerManifestFileName),
+    loadJsonFile(reactClientManifestFileName),
+  ]);
+
+  const ssrManifest = {
+    moduleLoading: {
+      prefix: '/webpack/development/',
+      crossOrigin: null,
+    },
+    moduleMap: {} as Record<string, unknown>,
+  };
+
+  Object.entries(reactClientManifest).forEach(([aboluteFileUrl, clientFileBundlingInfo]) => {
```

Comment:
Typo: `absoluteFileUrl` (spotted by CodeRabbit in https://github.com/shakacode/react_on_rails/pull/1696#pullrequestreview-2753463071, please check those comments).

Reply:
Also the casts below can be avoided if you cast either here or in `loadJsonFile(reactClientManifestFileName)`:
```suggestion
  (Object.entries(reactClientManifest) as Record<string, { id: string }>).forEach(([absoluteFileUrl, clientFileBundlingInfo]) => {
```